packages/flutter_tools/lib/src/platform_plugins.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+/// Marker interface for all platform specific plugin config impls.
+abstract class PlatformPlugin {
+  const PlatformPlugin();
+
+  Map<String, dynamic> toMap();
+}
+
+/// Contains parameters to template an Android plugin.
+///
+/// The required fields include: [name] of the plugin, [package] of the plugin and
+/// the [pluginClass] that will be the entry point to the plugin's native code.
+class AndroidPlugin extends PlatformPlugin {
+  const AndroidPlugin({
+    @required this.name,
+    @required this.package,
+    @required this.pluginClass,
+  });
+
+  factory AndroidPlugin.fromYaml(String name, dynamic yaml) {
```

Comment:
This should be a `YamlMap` to work right?

Reply:
It also works with `dynamic yaml`. Changing it to `YamlMap yaml` does make it better. Fixed!